src/ngx_http_lua_rewriteby.c

```diff
@@ -164,24 +171,73 @@ ngx_http_lua_rewrite_handler(ngx_http_request_t *r)
 
 
 ngx_int_t
-ngx_http_lua_rewrite_handler_inline(ngx_http_request_t *r)
+ngx_http_lua_rewrite_handler_sets(ngx_http_request_t *r)
 {
-    lua_State                   *L;
     ngx_int_t                    rc;
     ngx_http_lua_loc_conf_t     *llcf;
+    ngx_http_lua_ctx_t          *ctx;
+    ngx_uint_t                   i;
+    ngx_http_lua_phase_ctx_t    *phase_ctx;
 
-    dd("rewrite by lua inline");
+    dd("rewrite by lua handler sets");
+
+    rc = NGX_DECLINED;
+
+    ctx = ngx_http_get_module_ctx(r, ngx_http_lua_module);
+
+    dd("ctx = %p", ctx);
+
+    if (ctx == NULL) {
+        ctx = ngx_http_lua_create_ctx(r);
+        if (ctx == NULL) {
+            return NGX_HTTP_INTERNAL_SERVER_ERROR;
+        }
+    }
+
+    dd("entered rewrite by lua handler sets[%ld]", ctx->rewrite_index);
 
     llcf = ngx_http_get_module_loc_conf(r, ngx_http_lua_module);
+
+    phase_ctx = llcf->rewrites->elts;
+
+    for (i = ctx->rewrite_index; i < llcf->rewrites->nelts; ++i)
+    {
```

Comment:
Coding style: `{` should be in the previous line if the previous line is not splitted into multiple lines itself.